### ISSUE TITLE
Crash on Site Pages list.

### ISSUE THREAD

**Issue by aerych** [wordpress-mobile/WordPress-iOS#9182]
I'm seeing the following exception on the site pages list in the iPhone 8 plus simulator and on my iPhone 6s device.  This is occurring in Develop and in the 9.9 beta.

To reproduce: 
From a fresh install pull up My Sites > [Site] > Site Pages
Allow the list to refresh. 
Wait for it....
💥 

```
2018-04-24 10:26:28.583151-0500 WordPress[10056:233103] *** Terminating app due to uncaught exception 'Unresolved Core Data save error', reason: 'Unresolved Core Data save error - derived context'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000011438e1e6 __exceptionPreprocess + 294
	1   libobjc.A.dylib                     0x00000001138f1031 objc_exception_throw + 48
	2   WordPress                           0x00000001082d4e39 __57-[ContextManager saveDerivedContext:withCompletionBlock:]_block_invoke + 793
	3   CoreData                            0x000000011176d4e1 developerSubmittedBlockToNSManagedObjectContextPerform + 161
	4   libdispatch.dylib                   0x000000011542f848 _dispatch_client_callout + 8
	5   libdispatch.dylib                   0x000000011543a92b _dispatch_main_queue_callback_4CF + 628
	6   CoreFoundation                      0x0000000114350c99 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	7   CoreFoundation                      0x0000000114314ea6 __CFRunLoopRun + 2342
	8   CoreFoundation                      0x000000011431430b CFRunLoopRunSpecific + 635
	9   GraphicsServices                    0x0000000117c95a73 GSEventRunModal + 62
	10  UIKit                               0x000000010c0aa0b7 UIApplicationMain + 159
	11  WordPress                           0x00000001082d7b30 main + 144
	12  libdyld.dylib                       0x00000001154ac955 start + 1
	13  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by aerych** [comment]
Warnings from the console: 
```
2018-04-24 10:55:42.089408-0500 WordPress[11282:248985] [error] error:  API Misuse: Attempt to serialize store access on non-owning coordinator (PSC = 0x600000275100, store PSC = 0x0)
CoreData: error:  API Misuse: Attempt to serialize store access on non-owning coordinator (PSC = 0x600000275100, store PSC = 0x0)
2018-04-24 10:55:42.089570-0500 WordPress[11282:248985] [error] error:  API Misuse: Attempt to serialize store access on non-owning coordinator (PSC = 0x600000275100, store PSC = 0x0)
CoreData: error:  API Misuse: Attempt to serialize store access on non-owning coordinator (PSC = 0x600000275100, store PSC = 0x0)
2018-04-24 10:55:42.089693-0500 WordPress[11282:248985] [error] error:  API Misuse: Attempt to serialize store access on non-owning coordinator (PSC = 0x600000275100, store PSC = 0x0)
CoreData: error:  API Misuse: Attempt to serialize store access on non-owning coordinator (PSC = 0x600000275100, store PSC = 0x0)
2018-04-24 10:55:42.089816-0500 WordPress[11282:248985] [error] error:  API Misuse: Attempt to serialize store access on non-owning coordinator (PSC = 0x600000275100, store PSC = 0x0)
CoreData: error:  API Misuse: Attempt to serialize store access on non-owning coordinator (PSC = 0x600000275100, store PSC = 0x0)
2018-04-24 10:55:43.327100-0500 WordPress[11282:248985] *** Terminating app due to uncaught exception 'Unresolved Core Data save error', reason: 'Unresolved Core Data save error - derived context'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000114d7d1e6 __exceptionPreprocess + 294
	1   libobjc.A.dylib                     0x00000001142e8031 objc_exception_throw + 48
	2   WordPress                           0x0000000108bfde39 __57-[ContextManager saveDerivedContext:withCompletionBlock:]_block_invoke + 793
	3   CoreData                            0x00000001121c64e1 developerSubmittedBlockToNSManagedObjectContextPerform + 161
	4   libdispatch.dylib                   0x0000000115df6848 _dispatch_client_callout + 8
	5   libdispatch.dylib                   0x0000000115e0192b _dispatch_main_queue_callback_4CF + 628
	6   CoreFoundation                      0x0000000114d3fc99 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	7   CoreFoundation                      0x0000000114d03ea6 __CFRunLoopRun + 2342
	8   CoreFoundation                      0x0000000114d0330b CFRunLoopRunSpecific + 635
	9   GraphicsServices                    0x000000011865ca73 GSEventRunModal + 62
	10  UIKit                               0x000000010c9d30b7 UIApplicationMain + 159
	11  WordPress                           0x0000000108c00b30 main + 144
	12  libdyld.dylib                       0x0000000115e73955 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by frosty** [comment]
I spent some time looking into this, but I haven’t found the cause yet. I’m not seeing a crash either, but I am getting the Core Data errors logged out.

I can ‘fix’ the issue by ensuring we save the page list’s `managedObjectContext` before the fetched results controller performs a fetch. But that feels like a bit of a magic fix to me, and doesn’t resolve the underlying issue.

**Comment by aerych** [comment]
Looks like Media's `featuredOnPosts` <> `featuredImage` relation points to the `Post` model rather than `AbstractPost` where the inverse is defined.
<img width="885" alt="screen shot 2018-04-25 at 7 26 01 pm" src="https://user-images.githubusercontent.com/1435271/39279428-9cd33e38-48be-11e8-8309-c02a2c51f32a.png">

I'm not sure, but maybe we can safely update the relationship without anything blowing up? 😬

**Comment by frosty** [comment]
Thanks @aerych! I'll take another look today.

**Comment by frosty** [comment]
Fixed by #9209
